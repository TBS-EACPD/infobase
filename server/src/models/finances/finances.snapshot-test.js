const { execQuery } = global;

const org_vote_stat_pa_field = `
org_vote_stat_pa{
  vote_num
  vs_type
  name

  pa_last_year_5_auth
  pa_last_year_4_auth
  pa_last_year_3_auth
  pa_last_year_2_auth
  pa_last_year_auth

  pa_last_year_5_exp
  pa_last_year_4_exp
  pa_last_year_3_exp
  pa_last_year_2_exp
  pa_last_year_exp

  pa_last_year_5_unlapsed
  pa_last_year_4_unlapsed
  pa_last_year_3_unlapsed
  pa_last_year_2_unlapsed
  pa_last_year_unlapsed
}`;
const org_vote_stat_estimates_field = `
org_vote_stat_estimates{
  vote_num
  vs_type
  name
  doc

  est_last_year_4
  est_last_year_3
  est_last_year_2
  est_last_year
  est_in_year
}`;
const org_transfer_payments_field = `
org_transfer_payments{
  type
  name

  pa_last_year_5_auth
  pa_last_year_4_auth
  pa_last_year_3_auth
  pa_last_year_2_auth
  pa_last_year_1_auth

  pa_last_year_5_exp
  pa_last_year_4_exp
  pa_last_year_3_exp
  pa_last_year_2_exp
  pa_last_year_1_exp
}`;
const program_sobjs_field = `
program_sobjs{
  so_num
  pa_last_year_3
  pa_last_year_2
  pa_last_year
}`;
const program_vote_stat_field = `
program_vote_stat{
  vs_type
  pa_last_year_3
  pa_last_year_2
  pa_last_year
}`;
const program_spending_field = `
program_spending{
  pa_last_year_5_exp
  pa_last_year_4_exp
  pa_last_year_3_exp
  pa_last_year_2_exp
  pa_last_year_exp

  planning_year_1
  planning_year_1_rev
  planning_year_1_spa
  planning_year_1_gross

  planning_year_2
  planning_year_2_rev
  planning_year_2_spa
  planning_year_2_gross

  planning_year_3
  planning_year_3_rev
  planning_year_3_spa
  planning_year_3_gross
}`;
const program_fte_field = `
program_fte{
  pa_last_year_5
  pa_last_year_4
  pa_last_year_3
  pa_last_year_2
  pa_last_year
  pa_last_year_planned
  planning_year_1
  planning_year_2
  planning_year_3
}`;

const org_vote_stat_pa_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    org(dept_code: "CPCC") {
      ${org_vote_stat_pa_field}
    }
  }
}`;
const org_vote_stat_estimates_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    org(dept_code: "CPCC") {
      ${org_vote_stat_estimates_field}
    }
  }
}`;
const org_transfer_payments_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    org(dept_code: "CPCC") {
      ${org_transfer_payments_field}
    }
  }
}`;
const program_sobjs_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    program(id: "CPCC-ISC00") {
      ${program_sobjs_field}
    }
  }
}`;
const program_vote_stat_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    program(id: "CPCC-ISC00") {
      ${program_vote_stat_field}
    }
  }
}`;
const program_spending_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    program(id: "CPCC-ISC00") {
      ${program_spending_field}
    }
  }
}`;
const program_fte_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    program(id: "ND-ISC00") {
      ${program_fte_field}
    }
  }
}`;

describe("finance data", () => {
  it("org vote stat pa snapshot", async () => {
    const data = await execQuery(org_vote_stat_pa_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("org vote stat estimates snapshot", async () => {
    const data = await execQuery(org_vote_stat_estimates_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("org transfer payments snapshot", async () => {
    const data = await execQuery(org_transfer_payments_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("program sobjs snapshot", async () => {
    const data = await execQuery(program_sobjs_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("program vote stat snapshot", async () => {
    const data = await execQuery(program_vote_stat_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("program spending snapshot", async () => {
    const data = await execQuery(program_spending_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("program fte snapshot", async () => {
    const data = await execQuery(program_fte_query, {});
    return expect(data).toMatchSnapshot();
  });
});
