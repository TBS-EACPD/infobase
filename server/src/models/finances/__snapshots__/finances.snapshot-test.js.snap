// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finance data org transfer payments snapshot 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "org_transfer_payments": Array [
          Object {
            "name": "Contributions to support the advancement of Northern Science and Technology",
            "pa_last_year_1_auth": 9500621,
            "pa_last_year_1_exp": 9152366,
            "pa_last_year_2_auth": 8175000,
            "pa_last_year_2_exp": 5627081,
            "pa_last_year_3_auth": 1726972,
            "pa_last_year_3_exp": 1726971,
            "pa_last_year_4_auth": 0,
            "pa_last_year_4_exp": 0,
            "pa_last_year_5_auth": 0,
            "pa_last_year_5_exp": 0,
            "type": "c",
          },
          Object {
            "name": "Grants to support the advancement of Northern Science and Technology",
            "pa_last_year_1_auth": 10000,
            "pa_last_year_1_exp": 10000,
            "pa_last_year_2_auth": 157650,
            "pa_last_year_2_exp": 87305,
            "pa_last_year_3_auth": 10000,
            "pa_last_year_3_exp": 10000,
            "pa_last_year_4_auth": 0,
            "pa_last_year_4_exp": 0,
            "pa_last_year_5_auth": 0,
            "pa_last_year_5_exp": 0,
            "type": "g",
          },
          Object {
            "name": "Grants to individuals, organizations, associations and institutions to support research and activities relating to the polar regions",
            "pa_last_year_1_auth": 1546000,
            "pa_last_year_1_exp": 1545984,
            "pa_last_year_2_auth": 1398350,
            "pa_last_year_2_exp": 1388350,
            "pa_last_year_3_auth": 1076000,
            "pa_last_year_3_exp": 1076000,
            "pa_last_year_4_auth": 1086000,
            "pa_last_year_4_exp": 1086000,
            "pa_last_year_5_auth": 1086000,
            "pa_last_year_5_exp": 1086000,
            "type": "g",
          },
          Object {
            "name": "Contributions to individuals, organizations, associations and institutions to support research and activities relating to the polar regions",
            "pa_last_year_1_auth": 0,
            "pa_last_year_1_exp": 0,
            "pa_last_year_2_auth": 0,
            "pa_last_year_2_exp": 0,
            "pa_last_year_3_auth": 0,
            "pa_last_year_3_exp": 0,
            "pa_last_year_4_auth": 10000,
            "pa_last_year_4_exp": 10000,
            "pa_last_year_5_auth": 10000,
            "pa_last_year_5_exp": 10000,
            "type": "c",
          },
        ],
      },
    },
  },
}
`;

exports[`finance data org vote stat estimates snapshot 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "org_vote_stat_estimates": Array [
          Object {
            "doc": "ME",
            "est_in_year": 31704049,
            "est_last_year": 28089505,
            "est_last_year_2": 20963206,
            "est_last_year_3": 18853197,
            "est_last_year_4": 2434137,
            "name": "Operating and Program",
            "vote_num": "1",
            "vs_type": 1,
          },
          Object {
            "doc": "SEA",
            "est_in_year": 0,
            "est_last_year": -1704000,
            "est_last_year_2": 0,
            "est_last_year_3": 0,
            "est_last_year_4": 1,
            "name": "Operating and Program",
            "vote_num": "1",
            "vs_type": 1,
          },
          Object {
            "doc": "SEB",
            "est_in_year": 0,
            "est_last_year": 1,
            "est_last_year_2": 628559,
            "est_last_year_3": 0,
            "est_last_year_4": 0,
            "name": "Operating and Program",
            "vote_num": "1",
            "vs_type": 1,
          },
          Object {
            "doc": "SEC",
            "est_in_year": 0,
            "est_last_year": 0,
            "est_last_year_2": 0,
            "est_last_year_3": 0,
            "est_last_year_4": 8239414,
            "name": "Operating and Program",
            "vote_num": "1",
            "vs_type": 1,
          },
          Object {
            "doc": "VA",
            "est_in_year": 0,
            "est_last_year": 0,
            "est_last_year_2": 494718,
            "est_last_year_3": 362645,
            "est_last_year_4": 67190,
            "name": "Operating and Program",
            "vote_num": "1",
            "vs_type": 1,
          },
          Object {
            "doc": "SEC",
            "est_in_year": null,
            "est_last_year": null,
            "est_last_year_2": null,
            "est_last_year_3": null,
            "est_last_year_4": -6259365,
            "name": "Capital - Used until 2015-16",
            "vote_num": "7",
            "vs_type": 2,
          },
          Object {
            "doc": "VA",
            "est_in_year": null,
            "est_last_year": null,
            "est_last_year_2": null,
            "est_last_year_3": null,
            "est_last_year_4": 6259365,
            "name": "Capital - Used until 2015-16",
            "vote_num": "7",
            "vs_type": 2,
          },
          Object {
            "doc": "SEC",
            "est_in_year": null,
            "est_last_year": null,
            "est_last_year_2": null,
            "est_last_year_3": null,
            "est_last_year_4": -1716972,
            "name": "Grants & Contributions - Used until 2015-16",
            "vote_num": "9",
            "vs_type": 3,
          },
          Object {
            "doc": "VA",
            "est_in_year": null,
            "est_last_year": null,
            "est_last_year_2": null,
            "est_last_year_3": null,
            "est_last_year_4": 1716972,
            "name": "Grants & Contributions - Used until 2015-16",
            "vote_num": "9",
            "vs_type": 3,
          },
          Object {
            "doc": "ME",
            "est_in_year": 1097559,
            "est_last_year": 1017101,
            "est_last_year_2": 631025,
            "est_last_year_3": 622077,
            "est_last_year_4": 139948,
            "name": "Contributions to employee benefit plans",
            "vote_num": "S",
            "vs_type": 0,
          },
          Object {
            "doc": "SA",
            "est_in_year": 0,
            "est_last_year": 0,
            "est_last_year_2": 128049,
            "est_last_year_3": -118677,
            "est_last_year_4": 273930,
            "name": "Contributions to employee benefit plans",
            "vote_num": "S",
            "vs_type": 0,
          },
        ],
      },
    },
  },
}
`;

exports[`finance data org vote stat pa snapshot 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "org_vote_stat_pa": Array [
          Object {
            "name": "Operating/Program",
            "pa_last_year_2_auth": 19215842,
            "pa_last_year_2_exp": 15363078,
            "pa_last_year_2_unlapsed": null,
            "pa_last_year_3_auth": 1264603,
            "pa_last_year_3_exp": 1264603,
            "pa_last_year_3_unlapsed": null,
            "pa_last_year_4_auth": 2469953,
            "pa_last_year_4_exp": 2236843,
            "pa_last_year_4_unlapsed": null,
            "pa_last_year_5_auth": null,
            "pa_last_year_5_exp": null,
            "pa_last_year_5_unlapsed": null,
            "pa_last_year_auth": 22086483,
            "pa_last_year_exp": 20769053,
            "pa_last_year_unlapsed": null,
            "vote_num": "1",
            "vs_type": 1,
          },
          Object {
            "name": "Operating/Program",
            "pa_last_year_2_auth": null,
            "pa_last_year_2_exp": null,
            "pa_last_year_2_unlapsed": null,
            "pa_last_year_3_auth": 9476139,
            "pa_last_year_3_exp": 7896157,
            "pa_last_year_3_unlapsed": null,
            "pa_last_year_4_auth": null,
            "pa_last_year_4_exp": null,
            "pa_last_year_4_unlapsed": null,
            "pa_last_year_5_auth": null,
            "pa_last_year_5_exp": null,
            "pa_last_year_5_unlapsed": null,
            "pa_last_year_auth": null,
            "pa_last_year_exp": null,
            "pa_last_year_unlapsed": null,
            "vote_num": "3",
            "vs_type": 1,
          },
          Object {
            "name": "Capital",
            "pa_last_year_2_auth": null,
            "pa_last_year_2_exp": null,
            "pa_last_year_2_unlapsed": null,
            "pa_last_year_3_auth": 0,
            "pa_last_year_3_exp": 0,
            "pa_last_year_3_unlapsed": null,
            "pa_last_year_4_auth": null,
            "pa_last_year_4_exp": null,
            "pa_last_year_4_unlapsed": null,
            "pa_last_year_5_auth": null,
            "pa_last_year_5_exp": null,
            "pa_last_year_5_unlapsed": null,
            "pa_last_year_auth": null,
            "pa_last_year_exp": null,
            "pa_last_year_unlapsed": null,
            "vote_num": "7",
            "vs_type": 2,
          },
          Object {
            "name": "Grants & Contributions",
            "pa_last_year_2_auth": null,
            "pa_last_year_2_exp": null,
            "pa_last_year_2_unlapsed": null,
            "pa_last_year_3_auth": 0,
            "pa_last_year_3_exp": 0,
            "pa_last_year_3_unlapsed": null,
            "pa_last_year_4_auth": null,
            "pa_last_year_4_exp": null,
            "pa_last_year_4_unlapsed": null,
            "pa_last_year_5_auth": null,
            "pa_last_year_5_exp": null,
            "pa_last_year_5_unlapsed": null,
            "pa_last_year_auth": null,
            "pa_last_year_exp": null,
            "pa_last_year_unlapsed": null,
            "vote_num": "9",
            "vs_type": 3,
          },
          Object {
            "name": "Operating/Program",
            "pa_last_year_2_auth": null,
            "pa_last_year_2_exp": null,
            "pa_last_year_2_unlapsed": null,
            "pa_last_year_3_auth": null,
            "pa_last_year_3_exp": null,
            "pa_last_year_3_unlapsed": null,
            "pa_last_year_4_auth": null,
            "pa_last_year_4_exp": null,
            "pa_last_year_4_unlapsed": null,
            "pa_last_year_5_auth": 2449626,
            "pa_last_year_5_exp": 2445699,
            "pa_last_year_5_unlapsed": null,
            "pa_last_year_auth": null,
            "pa_last_year_exp": null,
            "pa_last_year_unlapsed": null,
            "vote_num": "25",
            "vs_type": 1,
          },
          Object {
            "name": "Contributions to employee benefit plans",
            "pa_last_year_2_auth": 503400,
            "pa_last_year_2_exp": 503400,
            "pa_last_year_2_unlapsed": null,
            "pa_last_year_3_auth": 413878,
            "pa_last_year_3_exp": 413878,
            "pa_last_year_3_unlapsed": null,
            "pa_last_year_4_auth": 118424,
            "pa_last_year_4_exp": 118424,
            "pa_last_year_4_unlapsed": null,
            "pa_last_year_5_auth": 144310,
            "pa_last_year_5_exp": 144310,
            "pa_last_year_5_unlapsed": null,
            "pa_last_year_auth": 759074,
            "pa_last_year_exp": 759074,
            "pa_last_year_unlapsed": null,
            "vote_num": "S",
            "vs_type": 999,
          },
        ],
      },
    },
  },
}
`;

exports[`finance data program fte snapshot 1`] = `
Object {
  "data": Object {
    "root": Object {
      "program": Object {
        "program_fte": Array [
          Object {
            "pa_last_year": 3281,
            "pa_last_year_2": 2870,
            "pa_last_year_3": 3198,
            "pa_last_year_4": 2967,
            "pa_last_year_5": 5765,
            "pa_last_year_planned": 3115,
            "planning_year_1": null,
            "planning_year_2": null,
            "planning_year_3": null,
          },
        ],
      },
    },
  },
}
`;

exports[`finance data program sobjs snapshot 1`] = `
Object {
  "data": Object {
    "root": Object {
      "program": Object {
        "program_sobjs": Array [
          Object {
            "pa_last_year": 2681685.4,
            "pa_last_year_2": 1462582.51,
            "pa_last_year_3": 846130.61,
            "so_num": 1,
          },
          Object {
            "pa_last_year": 433955.53,
            "pa_last_year_2": 637989.25,
            "pa_last_year_3": 269075.96,
            "so_num": 2,
          },
          Object {
            "pa_last_year": 22559.45,
            "pa_last_year_2": 32660.43,
            "pa_last_year_3": 32159.21,
            "so_num": 3,
          },
          Object {
            "pa_last_year": 534006.79,
            "pa_last_year_2": 756756.13,
            "pa_last_year_3": 420617.99,
            "so_num": 4,
          },
          Object {
            "pa_last_year": 1044006.95,
            "pa_last_year_2": 639710.32,
            "pa_last_year_3": 216099.23,
            "so_num": 5,
          },
          Object {
            "pa_last_year": 5176.8,
            "pa_last_year_2": 15078.78,
            "pa_last_year_3": 2161,
            "so_num": 6,
          },
          Object {
            "pa_last_year": 242665.09,
            "pa_last_year_2": 165902.85,
            "pa_last_year_3": 34223.83,
            "so_num": 7,
          },
          Object {
            "pa_last_year": null,
            "pa_last_year_2": 540689.82,
            "pa_last_year_3": null,
            "so_num": 8,
          },
          Object {
            "pa_last_year": 39230.35,
            "pa_last_year_2": 56318.91,
            "pa_last_year_3": 86965.04,
            "so_num": 9,
          },
          Object {
            "pa_last_year": 518.27,
            "pa_last_year_2": -11087.66,
            "pa_last_year_3": 564,
            "so_num": 12,
          },
        ],
      },
    },
  },
}
`;

exports[`finance data program spending snapshot 1`] = `
Object {
  "data": Object {
    "root": Object {
      "program": Object {
        "program_spending": Array [
          Object {
            "pa_last_year_2_exp": 4296601.34,
            "pa_last_year_3_exp": 1907996.87,
            "pa_last_year_4_exp": 305329.74,
            "pa_last_year_5_exp": 330773.1,
            "pa_last_year_exp": 5003804.63,
            "planning_year_1": null,
            "planning_year_1_gross": null,
            "planning_year_1_rev": null,
            "planning_year_1_spa": null,
            "planning_year_2": null,
            "planning_year_2_gross": null,
            "planning_year_2_rev": null,
            "planning_year_2_spa": null,
            "planning_year_3": null,
            "planning_year_3_gross": null,
            "planning_year_3_rev": null,
            "planning_year_3_spa": null,
          },
        ],
      },
    },
  },
}
`;

exports[`finance data program vote stat snapshot 1`] = `
Object {
  "data": Object {
    "root": Object {
      "program": Object {
        "program_vote_stat": Array [
          Object {
            "pa_last_year": 325642.67,
            "pa_last_year_2": 181727.57,
            "pa_last_year_3": 108394.94,
            "vs_type": "S",
          },
          Object {
            "pa_last_year": 4678161.96,
            "pa_last_year_2": 4114873.77,
            "pa_last_year_3": 1799601.93,
            "vs_type": "V",
          },
        ],
      },
    },
  },
}
`;
